lib/easypost/util.rb

```diff
@@ -79,14 +79,16 @@ def self.convert_to_easypost_object(response, api_key, parent=nil, name=nil)
       when Array
         return response.map { |i| convert_to_easypost_object(i, api_key, parent) }
       when Hash
-        if cls_name = response[:object]
-          cls = types[cls_name]
-        elsif response[:id] && cls_prefix = response[:id][0..response[:id].index('_')]
-          cls = prefixes[cls_prefix[0..-2]]
-        elsif response['id'] && cls_prefix = response['id'][0..response['id'].index('_')]
-          cls = prefixes[cls_prefix[0..-2]]
-        end
-
+        begin
+          if cls_name = response[:object]
+            cls = types[cls_name]
+          elsif response[:id] && cls_prefix = response[:id][0..response[:id].index('_')]
+            cls = prefixes[cls_prefix[0..-2]]
+          elsif response['id'] && cls_prefix = response['id'][0..response['id'].index('_')]
+            cls = prefixes[cls_prefix[0..-2]]
+          end
+        rescue ArgumentError => e
```

Comment:
I would prefer to handle this with a new check rather than a begin..rescue, since it's faster, clearer and less likely to cause side effects. I would simply add a new case that checks if `response[:id].index('_')` or `response['id'].index('_')` are non-nil before the second and third case that are there now. I'm happy to explain what I mean in more detail if the error or desired solution is unclear.

Reply:
I added a test and refactored util.rb to check for "_" in id before trying to convert to a specific easypost object.